### PR TITLE
Migrate CI Pipeline to CircleCI

### PR DESCRIPTION
**Pull request by ianstanton**
- Migrate CI pipeline from Drone to CircleCI
- Remove `.drone.yml`
- Pipeline steps and conditions should be virtually the same
- Change AWS region to `us-west-2`
- Related to astronomer/issues#2287,  astronomer/issues#882, astronomer/issues#930

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,142 @@
+version: 2.1
+orbs:
+  slack: circleci/slack@4.1.3
+jobs:
+  terraform_lint:
+    docker:
+      - image: quay.io/astronomer/ci-terraform:2020-12
+    steps:
+      - checkout
+      - run: cp providers.tf.example providers.tf
```
Comment (tommyk) [inline .circleci/config.yml]: could you potentially put all these "steps" into a bash script and just call it?  might be easier to edit, test, and understand.

thoughts?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+  terraform_apply:
+    docker:
+      - image: quay.io/astronomer/ci-terraform:2020-12
+    steps:
+      - checkout
+      - run: |
+            curl -o aws-iam-authenticator https://amazon-eks.s3-us-west-2.amazonaws.com/1.14.6/2019-08-22/bin/linux/amd64/aws-iam-authenticator
+            chmod +x ./aws-iam-authenticator
+            mv ./aws-iam-authenticator /usr/local/bin/
+            which aws-iam-authenticator
```
Comment (sjmiller609) [inline .circleci/config.yml]: optional: can this be added in the image quay.io/astronomer/ci-terraform:2020-12?

**Review by sjmiller609 — APPROVED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (danielhoherd) [reply .circleci/config.yml]: Absolutely. Wherever possible we should avoid installing third party software in CI runs. I will make that change now.

File: .circleci/config.yml (hunk 1 above)
Comment (danielhoherd) [reply .circleci/config.yml]: I definitely like the pattern of writing scripts and calling those vs embedding shell scripts in yaml. It is difficult to spot bugs in yaml embedded shell scripts. IE: we can't `shellcheck` yaml embedded shell scripts.

File: .circleci/config.yml (hunk 2 above)
Comment (danielhoherd) [reply .circleci/config.yml]: Done https://github.com/astronomer/ci-images/compare/268d78a1e..08dce78e

Comment (ianstanton) [reply .circleci/config.yml]: Good catch @sjmiller609. @danielhoherd Thanks for adding, I'll give it a test run

File: .circleci/config.yml (hunk 1 above)
Comment (ianstanton) [reply .circleci/config.yml]: @tommyk @danielhoherd Same thing we're doing with the `pipeline/run_terraform.sh` script in `terraform_apply` and `terraform_destroy` jobs? Should we use the same script or create a new one for these linting steps?

File: pipeline/run_terraform.sh [hunk 3]
```diff
@@ -1,23 +1,28 @@
 set -xe
 
-terraform -v
+TERRAFORM=terraform-0.12.29
```
Comment (danielhoherd) [inline pipeline/run_terraform.sh]: Doing it this way will let you specify TERRAFORM in the env.

```suggestion
TERRAFORM="${TERRAFORM:-terraform-0.12.29}"
```

File: .circleci/config.yml (hunk 1 above)
Comment (danielhoherd) [reply .circleci/config.yml]: I would say yes, just like that. Whether or not it's the same script isn't the important bit, so just do what makes sense. The important thing is that doing it as a script makes it cleaner, lintable, and portable.

File: pipeline/run_terraform.sh (hunk 3 above)
Comment (ianstanton) [reply pipeline/run_terraform.sh]: @danielhoherd Does this allow us to set that as a pipeline env var via the circle app? Does this default to `terraform-0.12.29`?

Comment (danielhoherd) [reply pipeline/run_terraform.sh]: Yes and yes

File: .circleci/config.yml [hunk 4]
```diff
@@ -7,6 +7,7 @@ jobs:
       - image: quay.io/astronomer/ci-terraform:2020-12
     steps:
       - checkout
+      - run: chmod +x pipeline/lint.sh
```
Comment (danielhoherd) [inline .circleci/config.yml]: This isn't needed. just chmod +x the file and commit that change to git.

**Review by danielhoherd — APPROVED** [review]
Looks pretty good. Three extra credit tasks would be:
- make all the shell scripts pass `shellcheck`
- add shellcheck and other useful hooks to `.pre-commit-config.yaml`
- run pre-commit as a step in CI

I don't think making those changes would take much effort, and the second 2 have reference code that you can copy/paste:
- https://github.com/astronomer/google-environments/blob/remove-prod-dynamic-pool/.circleci/config.yml#L12-L17
- https://github.com/astronomer/google-environments/blob/remove-prod-dynamic-pool/.pre-commit-config.yaml#L31-L34

If you want to pair on this to knock it out quickly we can do that.

**Comment by danielhoherd** [comment]
Oh, one more thing: i realized that specifying the full semver for terraform would become a problem so I updated the terraform ci image to also include `terraform-${major}.${minor}`, EG: `terraform-0.12`, so you could update that and that would make this more future proof.

**Review by danielhoherd — APPROVED** [review]
LGTM!

**Review by sunkickr — APPROVED** [review]
LGTM

**Pull request merged**